### PR TITLE
HL-275

### PR DESCRIPTION
**Pull request by brettvniekerk**
[Link to ticket](https://dltx.atlassian.net/jira/software/projects/HL/boards/82?selectedIssue=HL-275)

- Adds payout date check and makes bet pending if its not payable yet

### CODE REVIEW

File: src/components/Modals/SettleBetModal.tsx [hunk 1]
```diff
@@ -85,11 +90,7 @@ export const SettleBetModal: React.FC<Props> = ({
       ) : (
         <React.Fragment>
           <h2 className="font-bold text-2xl mr-[8vw] mb-6">
-            {selectedBet.settled || txHash
-              ? "Settled"
-              : selectedBet.winningPropositionId
-              ? "Unsettled"
-              : "Pending"}{" "}
+            {formatFirstLetterCapitalied(selectedBet.status)}{" "}
```
Comment (bazmatic) [inline src/components/Modals/SettleBetModal.tsx:93]:
```suggestion
            {formatFirstLetterCapitalized(selectedBet.status)}{" "}
```

Comment (brettvniekerk) [reply src/components/Modals/SettleBetModal.tsx]: done :D

File: src/utils/bets.ts [hunk 2]
```diff
@@ -17,6 +17,7 @@ const getBetStatus = (
 ): BetStatus => {
   const hasResult =
     signedBetData.winningPropositionId || signedBetData.marketResultAdded;
+  if (+bet.payoutAt > Math.floor(Date.now() / 1000)) return "PENDING";
   if (!hasResult && !bet.settled) return "PENDING";
```
Comment (bazmatic) [inline src/utils/bets.ts:21]: Better to use an enum

Comment (brettvniekerk) [reply src/utils/bets.ts:21]: these are string literals as part of the `BetStatus` type which @pondpiu wrote - i believe he had a reason for doing so

**Review by bazmatic — APPROVED** [review]

File: src/utils/formatting.ts [hunk 3]
```diff
@@ -76,3 +76,6 @@ export const parseBytes16String = (bytes: ethers.BytesLike) => {
     data.slice(0, nullTermination)
   );
 };
+
+export const formatFirstLetterCapitalised = (string: string) =>
+  `${string.charAt(0).toUpperCase()}${string.slice(1).toLowerCase()}`;
```
Comment (bitcoinbrisbane) [inline src/utils/formatting.ts:81]: May aswell write a unit test too

**Review by bitcoinbrisbane — CHANGES_REQUESTED** [review]
Lets unit test that function please

File: src/utils/formatting.ts (hunk 3 above)
Comment (brettvniekerk) [reply src/utils/formatting.ts:81]: done

**Review by JasminFarhi — APPROVED** [review]
LGTM

**Review by OptimusOpus — APPROVED** [review]

**Review by bitcoinbrisbane — APPROVED** [review]
LGTM

**Pull request merged**